streaming/src/test/scala/org/apache/gearpump/streaming/StreamingTestUtil.scala

```diff
@@ -37,4 +49,63 @@ object StreamingTestUtil {
     executorId += 1
     miniCluster.launchActor(props).asInstanceOf[TestActorRef[AppMaster]]
   }
+
+  def createEchoForTaskActor(taskClass: String, taskConf: UserConfig, system1: ActorSystem, system2: ActorSystem): (ActorRef, TestProbe) = {
+    import system1.dispatcher
+    val task1 = TaskDescription(taskClass, 1)
+    val task2 = TaskDescription(classOf[EchoTask].getName, 1)
+    val dag = DAG(Graph(task1 ~ new HashPartitioner() ~> task2))
+    val appId = 0
+    val taskReporter = TestProbe()(system2)
+    val taskId1 = TaskId(0, 0)
+    val taskId2 = TaskId(1, 0)
+    val executorId1 = 1
+    val executorId2 = 2
+    val appMaster = system1.actorOf(Props(classOf[MockAppMaster]))
+    val taskContext1 = TaskContext(taskId1, executorId1, appId, appMaster, dag)
+    val taskContext2 = TaskContext(taskId2, executorId2, appId, appMaster, dag)
+    val testActor = system1.actorOf(Props(Class.forName(taskClass), taskContext1, taskConf))
+    val reporter = system2.actorOf(Props(classOf[EchoTask], taskContext2, UserConfig.empty.withValue(EchoTask.TEST_PROBE, taskReporter.ref)))
+    val express1 = Express(system1)
+    val express2 = Express(system2)
+    val testActorHost = express1.localHost
+    val reporterHost = express2.localHost
+    val locations = Map((reporterHost, Set(taskId2)), (testActorHost, Set(taskId1)))
+    val result = locations.flatMap { kv =>
+      val (host, taskIdList) = kv
+      taskIdList.map(taskId => (TaskId.toLong(taskId), host))
+    }
+    val future = express1.startClients(locations.keySet).map { _ =>
```

Comment:
Have to say, this is tricky.

Can we do modifiction n taskactor or express to make them more testable?